sql/src/test/java/io/crate/integrationtests/JoinIntegrationTest.java

```diff
@@ -636,4 +636,13 @@ public void testAggOnJoinWithLimit() throws Exception {
                 "limit 0");
         assertThat(response.rowCount(), is(0L));
     }
+
+    @Test
+    public void testLimitIsAppliedPostJoin() throws Exception {
+        execute("select " +
+                "   sum(t1.col1) " +
+                "from unnest([1, 1]) t1, unnest([1, 1]) t2 " +
+                "limit 1");
+        assertThat(TestingHelpers.printedTable(response.rows()), is("4.0\n"));
```

Comment:
would be nice to add a test to check that the topn is still there e.g:

select ... limit 1 offset 1 -> no rows